src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,473 @@
+                "Strain": "",
+                "Isolate": "",
+                "Isolate/Strain": "",
+                "Asm name": "",
+                "Assembly type": "",
+                "Asm accession": "",
+                "Paired assembly": "",
+                "Asm last updated": "",
+                "Asm status": "",
+                "Asm notes": "",
+            }
+        )
+
+
+def examine_parameterization(
+    input_cores: PathLike, input_accessions: PathLike, db_host: str, db_port: int
+) -> Path:
+    """Detect the kind of user input (cores/accessions) and determine any missing or
+    incorrect parameterization.
+
+    Args:
+        input_cores: input core(s) list file name.
+        input_accessions: input accession (s) list file name.
+        db_host: host server name
+        db_port: host serer port
+
+    Returns:
+        User input file used in assembly status querying
+    """
+    user_query_file: Path
+
+    # Check for required input in the form of cores/accessions
+    if input_cores is None and input_accessions is None:
+        logging.critical(
+            "Missing required input: '--input_cores' (core db names) OR '--input_accns' (INSDC accessions)."
+        )
+        sys.exit()
+    elif input_cores and input_accessions:
+        logging.critical("Detected '--input_cores' AND '--input_accns'. Please provide just one such option.")
+        sys.exit()
+    # Input core names centered run
+    elif input_cores and input_accessions is None:
+        user_query_file = input_cores
+        logging.info(f"Performing assembly status report using core db list file: {user_query_file}")
+        if db_host is None or db_port is None:
+            logging.critical(
+                "User must specify both arguments '--host' and '--port' when providing core database names."
+            )
+            sys.exit()
+    # Accession centered run
+    elif input_cores is None and input_accessions:
+        user_query_file = input_accessions
+        logging.info(f"Performing assembly status report using INSDC accession list file: {user_query_file}")
+
+    return user_query_file
+
+
+def resolve_query_type(
+    query_list: list, host_server: str, host_port: str, input_cores: str, input_accessions: str
+) -> Union[Tuple[Dict, str]]:
+    """Function to indentify the kind of querys being passed by user,
+    then extract the queries (core names or accesisons) and store each with appropriate identifier.
+
+    Args:
+        query_list: List of user defined queries either core names, or accessions
+        input_cores: arg parse param '--input_cores'
+        input_accessions: arg parse param '--input_accns'
+
+    Returns:
+        User queries stored as indentifier[(core db name | UniqueID#)] : accession
+    """
+
+    query_accessions: Dict = {}
+    query_type: str = ""
+
+    if input_cores and input_accessions is None:
+        server_details = f"mysql://ensro@{host_server}:{host_port}/"  ## Requires some more dev !!
```

Comment:
You could use `URL` here:
```
from sqlalchemy.engine import URL
test_url = URL.create(server_details)
```
Or better yet, provide this URL object to the function (prepare it in the main) instead of `host_server` and `host_port`.

Reply:
Nice, update. Thank you